susemanager-utils/susemanager-sls/src/modules/mgrnet.py

```diff
@@ -0,0 +1,137 @@
+"""
+Module for gathering DNS FQDNs
+"""
+
+import logging
+import re
+import time
+from threading import Thread
+from queue import Queue
+
+import salt.utils.network
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    """
+    Only works on POSIX-like systems having 'host' or 'nslookup' available
+    """
+    if not (__utils__["path.which"]("host") or __utils__["path.which"]("nslookup")):
+        log.debug("Neither 'host' nor 'nslookup' is available on the system")
+    return True
+
+
+class ThreadPool:
```

Comment:
Could you share your reasoning behind having this `ThreadPool`? What's the advantage over using something simple like:
```python
ips = ["1.1.1.1", "2.2.2.2", "3.3.3.3"]
threads = []
for ip in ips: # I didn't account for an upper limit to keep this brief, adding it is simple
    t = threading.Thread(target=lookup_dns_fqdn, args=(ip,))
    threads.append(t)
    t.start()

for t in threads:
    t.join()
```

Alternatively, have you considered using https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor instead of our own implementation?

Reply:
There could be significant amount of attached interfaces on the client. Not really usual, but some of the customers have more than 10 network interfaces, so it was made just to limit the possible large list of IPs. I'll check with the docs of ThreadPoolExecutor. Thanks for pointing to it :+1:

Reply:
As I wrote in the code comment, instead of using `for ip in ips` we can use a way that has an upper limit. That was not the point if the suggestion, the point was to create the X number of threads in a simple loop instead of having a class that abstracts it away.
```python
for i in range(min(8, len(ips))):
    ip = ips[i]
...
```